pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/creator/JsonIndexCreator.java

```diff
@@ -29,7 +29,39 @@
   char KEY_VALUE_SEPARATOR = '\0';
 
   /**
-   * Adds the next json value.
+   * To be invoked once before each physical JSON document when streaming JSON into the index
+   * @param numFlattenedDocs the number of flattened documents to be streamed into the index
+   */
+  default void startDocument(int numFlattenedDocs) {
+  }
+
+  /**
+   * To be invoked after each JSON document when streaming JSON into the index.
+   */
+  default void endDocument() {
+  }
+
+  /**
+   * To be invoked once before each flattened document when streaming JSON into the index
+   */
+  default void startFlattenedDocument() {
+  }
+
+  /**
+   * To be invoked once after each flattened JSON document when streaming JSON into the index
+   */
+  default void endFlattenedDocument() {
```

Comment:
Interface should not have the concept of "flattened" documents as this won't apply to a Tree-based JSON index of maybe even a hash-based JSON index.